### ISSUE TITLE
[Feature Request] key-holding and lock aspect-ratio

### ISSUE THREAD

**Issue by Spidy123222** [utmapp/UTM#532]
**Key-Holding:**

I think what would be useful for the people who actually got games working on UTM that don't need GPU acceleration to be able to hold keys on a keyboard or able to setup a custom keyboard to easily access keys from top bar like macros in a way.

For example if I want to charge my gun in _Worms Armageddon_ I would need to hold space bar but it's currently impossible to do so.

**Locking Aspect-Ratio:**

By locking aspect ratio you can change you're device orientation and keep the aspect ratio so it doesn't break UI elements on anything that has a little graphics. Most programs don't support 9:16 but they support 16:9 or even 4:3.

**Comment by fomalsd** [comment]
use hardware keyboard or a gamepad, playing games with virtual keyboard occupying half the screen isn't much fun anyway

**Comment by Spidy123222** [comment]
> use hardware keyboard or a gamepad, playing games with virtual keyboard occupying half the screen isn't much fun anyway

I thought virtual keyboard was alright. Maybe if we had a separate keyboard for macros would be nice. Sadly I had neither game pad or hardware keyboard that support iOS.

**Comment by IIWare** [comment]
@Spidy123222 can't you hold the keys manually with your fingers?

**Comment by Spidy123222** [comment]
> @Spidy123222 can't you hold the keys manually with your fingers?

I can but it doesn't act as a hold and holding only works on the top bar buttons wich are buttons from UTm so if having a custom keyboard or onscreen buttons to press would making key holding or using certain keys for stuff. Does that make sense?

**Comment by osy86** [comment]
I think it works with the custom keystrip but not with the iOS on screen keyboard. I don't think it's possible to detect held down keys with the OSK.

**Comment by Spidy123222** [comment]
> I think it works with the custom keystrip but not with the iOS on screen keyboard. I don't think it's possible to detect held down keys with the OSK.

Yes this is what I'm seeing happening. Did you happen to check out the video to see?

**Comment by conath** [comment]
Indeed there's no way for an app to detect the iOS software keyboard key "hold" events. Here's what would be optimal: I'd suggest we make a configurable UTM gaming keyboard that can be activated as an alternative to the iOS software keyboard. It would have hold support for any button, and maybe include a joystick/dpad for mouse movements. I'm theoretically down for making that happen but I'm not sure when I will have the time.

**Comment by DougJoseph** [comment]
I don't know if I need to post a new issue or if this is the right place.

For host I'm running an M1 Max (MacBook Pro), and for guest, Windows XP. 

The expected behavior is when I hold down any key the action is causes gets repeated. For example, if I hold down a cursor navigation key, it should keep going that direction, not go only one notch and stop.

What actually happens, is that it takes the cursor over one space or place, and stops. This happens with all keys.

I looked in UTM preferences, and under Input, I found nothing related to this.

Am I missing something?
